include/verilated_types.h

```diff
@@ -172,18 +173,43 @@ class VlEvent final {
     VlEvent() = default;
     ~VlEvent() = default;
 
+    friend std::string VL_TO_STRING(const VlEvent& e);
+#ifdef VL_EVENT_ASSIGN
+    friend class VlEventHandle;
+#else
     // METHODS
     void fire() { m_fired = m_triggered = true; }
     bool isFired() const { return m_fired; }
     bool isTriggered() const { return m_triggered; }
     void clearFired() { m_fired = false; }
     void clearTriggered() { m_triggered = false; }
+#endif
 };
 
 inline std::string VL_TO_STRING(const VlEvent& e) {
-    return std::string{"triggered="} + (e.isTriggered() ? "true" : "false");
+    return std::string{"triggered="} + (e.m_triggered ? "true" : "false");
 }
 
+#ifdef VL_EVENT_ASSIGN
+class VlEventHandle final : public std::shared_ptr<VlEvent> {
```

Comment:
This means the main library can't be used with a design that uses event assignments, and another that does not, even in e.g. two different contexts.

Can we instead make a base class that can have either representations and have the emitted C++ code choose appropriately?

Reply:
Ok, makes sense. I didn't realize the same header would be included for different models.

I changed the solution so there are two different types which coexist in the header. Still, only one is used for a given model. They also share a common base class which is used to pass events to tracing functions.